### ISSUE TITLE
enable torch module multiple outputs

### ISSUE THREAD

**Issue by sheercyy** [dotnet/TorchSharp#697]
Hi there, seems that current version can only allow single output from the torch module. Is it possible to enable multiple outputs? A python analogy of doing this in Pytorch is here: https://discuss.pytorch.org/t/a-model-with-multiple-outputs/10440. This would be important to train a model with multiple loss functions , e.g., a variational autoencoder.

**Comment by kaiidams** [comment]
Some of TochSharp modules output multiple tensors like this using `new` modifier. Hope this helps.

https://github.com/dotnet/TorchSharp/blob/c7f9a2f8566d1f937e69c026708246b05c2d8d18/src/TorchSharp/NN/Recurrent/GRU.cs#L38
```
            public new (Tensor, Tensor) forward(Tensor input, Tensor h0 = null)
```

**Comment by sheercyy** [comment]
Thanks for the reply, it really helps a lot.

> Some of TochSharp modules output multiple tensors like this using `new` modifier. Hope this helps.
> 
> https://github.com/dotnet/TorchSharp/blob/c7f9a2f8566d1f937e69c026708246b05c2d8d18/src/TorchSharp/NN/Recurrent/GRU.cs#L38
> 
> ```
>             public new (Tensor, Tensor) forward(Tensor input, Tensor h0 = null)
> ```
